tests/unit/policyEvaluator.js

```diff
@@ -0,0 +1,930 @@
+'use strict'; // eslint-disable-line strict
+
+const assert = require('assert');
+
+const evaluator = require('../../lib/policyEvaluator/evaluator.js');
+const evaluatePolicy = evaluator.evaluatePolicy;
+const evaluateAllPolicies = evaluator.evaluateAllPolicies;
+const handleWildcards =
+    require('../../lib/policyEvaluator/utils/wildcards.js').handleWildcards;
+const substituteVariables =
+    require('../../lib/policyEvaluator/utils/variables.js');
+const samples = require('../utils/samplePolicies.json');
+const DummyRequestLogger = require('./helpers').DummyRequestLogger;
+
+const log = new DummyRequestLogger();
+
+
+function check(requestContext, policy, expected) {
+    const result = evaluatePolicy(requestContext, policy, log);
+    assert.deepStrictEqual(result, expected);
+}
+let policy;
+let requestContext;
+
+describe('policyEvaluator', () => {
+    describe('evaluate a single policy', () => {
+        describe('with basic checks', () => {
+            beforeEach(() => {
+                requestContext = {
+                    action: 's3:ListBucket',
+                    resource: 'arn:aws:s3:::superbucket',
+                    requesterInfo: {},
+                    headers: {},
+                    query: {},
+                };
+            });
+            it('should permit access under full access policy', () => {
+                check(requestContext,
+                    samples['arn:aws:iam::aws:policy/AmazonS3FullAccess'],
+                    'Allow');
+            });
+
+            it('should permit access under read only policy', () => {
+                check(requestContext,
+                    samples['arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess'],
+                    'Allow');
+            });
+
+            it('should be neutral under multi-statement ' +
+                'policy if request resource not in policy',
+                () => {
+                    requestContext.action = 's3:PutObject';
+                    check(requestContext,
+                        samples['Multi-Statement Policy'],
+                        'Neutral');
+                });
+
+            it('should grant access under multi-statement ' +
+                'policy if action and resource match those in policy',
+                () => {
+                    requestContext.action = 's3:GetObject';
+                    requestContext.resource =
+                        'arn:aws:s3:::notVeryPrivate/object';
+                    check(requestContext,
+                        samples['Multi-Statement Policy'],
+                        'Allow');
+                });
+
+            it('should be neutral under multi-statement ' +
+                'policy if request action not in policy',
+                () => {
+                    requestContext.action = 's3:DeleteObject';
```

Comment:
I tried to change them where they made sense.  For instance, checking if an action is not in a policy, I had to change the action.  Also, if conditions did not make sense for an action, I changed it (such as having a condition with maxKeys, etc. with a listing action).